st2client/st2client/commands/resource.py

```diff
@@ -731,20 +732,47 @@ def __init__(self, resource, *args, **kwargs):
 
         self.parser.add_argument(argument, metavar=metavar, help=help)
 
+        self.parser.add_argument(
+            "--yes",
+            action="store_true",
+            help="Auto yes flag to delete action files from disk.",
+        )
+
     @add_auth_token_to_kwargs_from_cli
     def run(self, args, **kwargs):
         resource_id = getattr(args, self.pk_argument_name, None)
         instance = self.get_resource(resource_id, **kwargs)
-        self.manager.delete(instance, **kwargs)
+        if args.yes:
+            self.manager.delete(instance, **kwargs)
+            print(
+                'Resource with id "%s" has been successfully deleted from database and disk.'
+                % (resource_id)
+            )
+        else:
+            if isinstance(instance, st2client.models.action.Action):
+                user_input = input(
+                    "It will delete action files on disk as well. Do you want to continue? (y/n): "
+                )
+                if user_input.lower() == "y" or user_input.lower() == "yes":
+                    self.manager.delete(instance, **kwargs)
+                    print(
+                        'Resource with id "%s" has been successfully deleted from db and disk.'
+                        % (resource_id)
+                    )
+                else:
+                    print("Action is not deleted.")
+            else:
+                self.manager.delete(instance, **kwargs)
+                print(
+                    'Resource with id "%s" has been successfully deleted.'
+                    % (resource_id)
+                )
```

Comment:
By putting this in the ResourceDeleteCommand, this will affect other pack contents (not just action) and expanded the scope for this PR. So will you be expanding the scope?  If so, please make sure pack contents such as rules, policies, etc. are going to be deleted from disk properly as well.

Reply:
@m4dcoder 
Moving code under ResourceDeleteCommand class at /st2client/commands/resource.py to ActionDeleteCommand class at https://github.com/StackStorm/st2/blob/master/st2client/st2client/commands/action.py#L280 causes problems like if we import code from action.py to resource.py it gives import errors for other modules, it is probably because of circular imports. It becomes complex to call method from child class to parent class. Also, there is code under ResourceDeleteCommand on which test cases are dependent.

As we are handling condition if isinstance(instance, Action) therefore as per my understanding we are not affecting other pack contents like rules, policies etc. and if the resource is not instance of Action then flow is exactly same as it was before making the changes.

But still if this code need to be moved, please suggest me another appropriate place to move code under ResourceDeleteCommand to that location. Thanks.

Reply:
We shouldn't be handling the condition for sub type using if isinstance(instance, Action) in a generic class. The ActionDeleteCommand is there to override logic for specific sub type. I advise that you revisit putting this under ActionDeleteCommand. Can you post the import errors here?

Reply:
I realized the mistake I was doing. I was creating another function under ActionDeleteCommand and importing it in resource.py to use under ResourceDeleteCommand. Now I have shifted the whole code along with run and print_and_run methods required to delete action files under ActionDeleteCommand. Which is working fine.